packages/full-stack-tests/src/hierarchy-builder/HierarchyLevelFiltering.test.ts

```diff
@@ -0,0 +1,371 @@
+          relatedInstances: [],
+          rules: {
+            propertyName: `CodeValue`,
+            operator: "Like",
+            value: `test%`,
+          },
+        },
+      });
+      validateHierarchyLevel({
+        nodes,
+        expect: [
+          NodeValidators.createForInstanceNode({ instanceKeys: [keys.childSubject1] }),
+          NodeValidators.createForInstanceNode({ instanceKeys: [keys.childSubject2] }),
+        ],
+      });
+    });
+
+    it("filters child hierarchy level", async function () {
+      const { imodel, ...keys } = await buildIModel(this, async (builder) => {
+        const rootSubject = { className: subjectClassName, id: IModel.rootSubjectId };
+        const childSubject1 = insertSubject({ builder, codeValue: "test subject 1", parentId: rootSubject.id });
+        const childSubject2 = insertSubject({ builder, codeValue: "test subject 2", parentId: rootSubject.id });
+        return { rootSubject, childSubject1, childSubject2 };
+      });
+
+      const selectQueryFactory = new NodeSelectQueryFactory(createMetadataProvider(imodel));
+      const hierarchy: IHierarchyLevelDefinitionsFactory = {
+        async defineHierarchyLevel({ instanceFilter }) {
+          const filterClauses = instanceFilter
+            ? await selectQueryFactory.createFilterClauses(instanceFilter, { fullName: subjectClassName, alias: "this" })
+            : undefined;
+          return [
+            {
+              fullClassName: subjectClassName,
+              query: {
+                ecsql: `
+                  SELECT ${await selectQueryFactory.createSelectClause({
+                    ecClassId: { selector: `this.ECClassId` },
+                    ecInstanceId: { selector: `this.ECInstanceId` },
+                    nodeLabel: { selector: `this.CodeValue` },
+                  })}
+                  FROM ${filterClauses?.from ?? subjectClassName} AS this
+                  ${filterClauses?.joins ?? ""}
+                  WHERE
+                    this.Parent.Id = 0x1
+                    ${filterClauses?.where ? `AND ${filterClauses?.where}` : ""}
+                `,
+              },
+            },
+          ];
+        },
+      };
+
+      const provider = createProvider({ imodel, hierarchy });
+      const nodes = await provider.getNodes({
+        parentNode: {
+          key: { type: "instances", instanceKeys: [{ className: subjectClassName, id: "0x1" }] },
+          parentKeys: [],
+          label: "",
+        },
+        instanceFilter: {
+          propertyClassName: subjectClassName,
+          relatedInstances: [],
+          rules: {
+            propertyName: `CodeValue`,
+            operator: "Like",
+            value: `%1`,
+          },
+        },
+      });
+      validateHierarchyLevel({
+        nodes,
+        expect: [NodeValidators.createForInstanceNode({ instanceKeys: [keys.childSubject1] })],
+      });
+    });
+
+    it("filters by property class", async function () {
```

Comment:
It is very difficult to understand what is happening in this test. We insert one `PhysicalPartition` element and then validating that with filter applied we get node for that element. It looks like without filter we also would get one node for that element. 

I guess it would help if hierarchy level without filter would be validated too.